### ISSUE TITLE
Can the `initialValue` of `useForm` be populated with a ref?

### ISSUE THREAD

**Issue by johnson86tw** [Mini-ghost/vorms#23]
### Describe the feature

Greetings,

I am currently implementing a scenario using this package. In the case of searching for a product by name, I would like to have my reactive URL, reactive form, and reactive API request all share the same `ref`. The code structure would look something like this:

```ts
// Updates the URL synchronously
const { search } = useRouteQuery("search") 

// Validates the input
const { values } = useForm({ 
    initialValues: {
        search: search,
    }
}

// Allows reusability of this composable in other components
const { products, pending, error, refresh } = await useProducts({ 
    params: {
        search: search
    }
})
```

When designing the `useProducts` function, I modified the type of `useProductsOption.params.search` to `type MaybeRef<T> = T | Ref<T>`, allowing it to accept a ref and share the same set of ref variables. I would like to inquire whether the `initialValues` of `useForm` can achieve this functionality.

Perhaps there is a better way to solve my issue of sharing the same ref? I would like to hear your suggestions. 

Cheers

### Additional information

- [X] Would you be willing to help implement this feature?

**Comment by chnejohnson** [comment]
So, if I understand correctly, you're suggesting that I can skip using the `useForm` composable and directly bind the "search" with the input field. However, in case I want to perform validation on the "search" input, I'm unsure if it fits the use case of Vorms in this context, where it's not about submitting a form but rather implementing a search and filtering functionality. What is your opinion on the matter?

**Comment by Mini-ghost** [comment]
Yes, based on the scenario you provided, using Vorms may not make your problem simpler.

Especially in your example:

```ts
const { values } = useForm({ 
    initialValues: {
        // search is a `Ref`
        search: search,
    }
}
```

If `search` is a `Ref` and it is only used as a property of `initialValues` rather than `initialValues` itself, I think I would need to check each property individually during initialization, which might make the situation more complicated, and there may be issues with `resetForm` as it currently stands.

If `search` is a `Reactive` or `Ref` and the root is an object, there is a greater possibility that it will work in the idea I proposed, but I still need to write some test cases to verify.

Additionally, if you want to perform validation while typing, you can use it like this:

```ts
const { register } = useForm({
  initialValues: searchParams,
  validateMode: 'change', // or 'input'
})

// need to bind value and attributes on the input field
const { value, attrs } = register('keyword')
```

However, it still requires waiting for Vorms to support passing a `Reactive` or `Ref` as `initialValues` in order to solve your problem.

I apologize if I didn't directly solve your problem, but I appreciate the suggestions you provided as they gave me some interesting ideas.

**Comment by chnejohnson** [comment]
Regarding the idea you presented, I believe I may require a more flexible design to separately define the reactive route query and useForm's initialValues. For instance, when it comes to the "page" parameter, I do not need to validate it, so the structure of routeQuery might differ from that of initialValues:

```ts
// This is a `Reactive`
const routeQuery = useRouteQuery({
        search: route.query.search,
        page: route.query.page,
})

const { values } = useForm({
      initialValues: {
              search: toRef(routeQuery, "search"),
      }
})

const { products, pending, error, refresh } = await useProducts({
        params: {
              search: toRef(routeQuery, "search"),
              page: toRef(routeQuery, "page"),
        }
})
```

No worries, I may allocate some time to study this functionality.

**Comment by Mini-ghost** [comment]
Thank you for your assistance, and I'm looking forward to the results of your study as well! 💚

Additionally, I might still suggest directly passing `routeQuery` to Vorms' `initialValues`. We can skip the validation rules for properties like `page` or `perPage` when writing the validation rules. This way, we won't need to create new `Ref` using `toRef`.

**Comment by chnejohnson** [comment]
Indeed, I am also uncertain whether using excessive `toRef` calls for multiple parameters could potentially lead to performance bottlenecks. However, the advantage of splitting them is that some parameters requiring validation might not come from `routeQuery` but rather from other variables defined within the page

**Comment by chnejohnson** [comment]
@Mini-ghost Hi, I kindly request your assistance in troubleshooting the problems I encountered while running `pnpm build:core`. Currently, my objective is to make code alterations and experiment with the new functionalities in the `/playground`. Would it be necessary to perform a build first in order to observe these changes manifesting in the playground?

![截圖 2023-05-28 上午10 02 27](https://github.com/Mini-ghost/vorms/assets/31992185/cf2122cc-a4b2-437b-ad67-28496a13bad6)

**Comment by Mini-ghost** [comment]
@chnejohnson Can you please provide the version of Vue you are currently using? You can execute `pnpm why vue` to confirm.

**Comment by chnejohnson** [comment]
Sure
<img width="760" alt="截圖 2023-05-29 下午7 22 45" src="https://github.com/Mini-ghost/vorms/assets/31992185/46b21cbb-ece8-4cd7-86aa-5e73e276ef03">

**Comment by Mini-ghost** [comment]
@chnejohnson There doesn't seem to be any issue. I've tried several different environments here and couldn't reproduce the problem. I'm not sure what the reason could be. Have you considered cloning the repository again to give it a try?

**Comment by chnejohnson** [comment]
Sorry...reinstalling solved the problem. Thanks for your help
